libp2p/protocols/kademlia/find.nim

```diff
@@ -139,28 +163,30 @@ proc findPeer*(
   return ok(PeerInfo(peerId: peer, addrs: kad.switch.peerStore[AddressBook][peer]))
 
 proc findClosestPeers*(kad: KadDHT, target: Key): seq[Peer] =
+  let closerPeers = kad.rtable.findClosest(target, kad.config.replication).filterIt(
+      it != kad.switch.peerInfo.peerId.toKey()
+    )
+
   var closestPeers: seq[Peer]
-  let selfKey = kad.switch.peerInfo.peerId.toKey()
-  for p in kad.rtable.findClosest(target, kad.config.replication):
-    if p == selfKey: # do not return self as one of closest peers
-      continue
-    let peer = p.toPeer(kad.switch).valueOr:
-      continue
-    closestPeers.add(peer)
+
+  for p in closerPeers:
+    p.toPeer(kad.switch).withValue(peer):
+      closestPeers.add(peer)
+
   return closestPeers
```

Comment:
add utility `toPeer` (or however you would call this transformation in `for`):

```toPeer(peer seq[????]) seq[Peer]```

then 

`return toPeer(closestPeers)`